### DISCUSSION TITLE
add terraform snippets

### DISCUSSION THREAD

**Discussion by armyers** [LazyVim/LazyVim#2114] [Q&A]
Hi,

I'm trying to enable terraform snippets. I see the file `.local/share/nvim/lazy/friendly-snippets/snippets/terraform.json` but there's no entry for the terraform language in the `friendly-snippets/package.json` file. If I hand-bomb package.json and add

```json
            {
                "language": "terraform",
                "path": "./snippets/terraform.json"
            }
```

Then I see snippets when editing a terraform file. But I know that's not the right way to do it.

I've removed that entry from package.json and created a plugins file for luasnip in `.config/nvim/lua/plugins/luasnip.lua:

```lua
return {
  {
    "L3MON4D3/LuaSnip",
    dependencies = {
      "rafamadriz/friendly-snippets",
      config = function()
        require("luasnip").filetype_extend("terraform", { "terraform" })
        require("luasnip.loaders.from_vscode").lazy_load()
      end,
    },
  },
}
```

but I'm not getting any terraform snippets.

Please help! :)

**Comment by dpetka2001** [comment]
You have to have a `package.json` if you use `vscode` loader. It's mentioned in the Luasnip [docs](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#loaders).

**Reply by armyers** [reply]
Hi,

sorry, I'm still confused. I've read the luasnip docs you referenced, but I still have a Q: are you thinking that I'm trying to load my own custom snippets?

If so, that's not what I'm attempting. I'm using lazyvim with luasnip, and it's loading friendly-snippets. The friendly-snippets plugin has this file of terraform snippets:  https://github.com/rafamadriz/friendly-snippets/blob/main/snippets/terraform.json

But, I think the terraform snippets are not being loaded because they are not included in the package.json that comes with the friendly-snippets plugin by default: https://github.com/rafamadriz/friendly-snippets/blob/main/package.json

And what I tried first is to edit (hand-bomb) the package.json file to include the terrafrom.json file, and that worked. But it doesn't seem like the best/right way to accomplish my goal.

So I'm trying to figure out the config code that I need in order to load that terraform.json file.

Thanks again for your help!

**Reply by dpetka2001** [reply]
> sorry, I'm still confused. I've read the luasnip docs you referenced, but I still have a Q: are you thinking that I'm trying to load my own custom snippets?

I'm teribbly sorry. That was exactly what I thought. I guess I should have read more carefully. Maybe create an issue at the repo to fix that then? That would certainly be the correct way I would think.

**Reply by armyers** [reply]
ah, got it. thank you again for your help! :)

**Reply by armyers** [reply]
and someone else has the same problem. Opened just 40 mins ago:  https://github.com/rafamadriz/friendly-snippets/issues/385
